aptos-move/aptos-vm/src/zkid_validation.rs

```diff
@@ -38,74 +46,67 @@ fn get_jwks_onchain(resolver: &impl AptosMoveResolver) -> anyhow::Result<Patched
     Ok(jwks)
 }
 
+fn get_jwk_for_zkid_authenticator(
+    jwks: &PatchedJWKs,
+    zkid_pub_key: &ZkIdPublicKey,
+    zkid_sig: &ZkIdSignature,
+) -> anyhow::Result<JWK, VMStatus> {
+    let jwt_header_parsed = zkid_sig
+        .parse_jwt_header()
+        .map_err(|_| invalid_signature!("Failed to get JWT header"))?;
+    let jwk_move_struct = jwks
+        .get_jwk(&zkid_pub_key.iss, &jwt_header_parsed.kid)
+        .map_err(|_| invalid_signature!("JWK not found"))?;
+
+    let jwk =
+        JWK::try_from(jwk_move_struct).map_err(|_| invalid_signature!("Could not parse JWK"))?;
+    Ok(jwk)
+}
+
 pub fn validate_zkid_authenticators(
```

Comment:
So we also have https://github.com/aptos-labs/aptos-core/blob/b40329261475357cc4527e65668f198aa2787fc9/aptos-move/aptos-vm/src/aptos_vm.rs#L2248, but it seems nothing is verified there and just flags are checked. Is this intentional?

Reply:
Yes. We do all our flag checking there.  I agree it's not ideal.

Here's [the thread](https://github.com/aptos-labs/aptos-core/pull/11681#discussion_r1458192227) about this from the other pr

tldr - it's an annoying refactor.  We'd need to update validate_signed_transaction to take in Vec<SingleKeyAuthenticator> or move all the feature flag checking into validate_signed_transaction.

down to do it if you have a preference

Reply:
yes, refactoring is annoying. Let's add a TODO for now so we do not forget, and revisit as a part of more substantial VM refactoring